### PR TITLE
Deropapy

### PR DESCRIPTION
**Pull request by HamishBrownPFR**
Working on #8175 #7869 #7075 (suite of simple crop, tree and pasture models)

Added fRadnAllZones variable into ICanopy.  This is the proportion of the total radiation over both zones that a crops in a specific zone is intercepting.  We have been calculating this in manager scripts but seems safer to do it in Microclimate

Some harmonizing of approaches and functionality between SCRUM, STRUM and SPRUM

### CODE REVIEW

**Comment by HamishBrownPFR** [comment]
@hol353 @par456 @ric394  can someone please review and merge if all OK

**Comment by HamishBrownPFR** [comment]
@hol353 @par456 @ric394 can someone please review and merge if all OK

**Comment by HamishBrownPFR** [comment]
@hol353 @par456 @ric394 can someone please review and merge if all OK or let me know what needs changing if anything

**Comment by hol353** [comment]
@hut104 and @HamishBrownPFR: Hamish we hear you but Neil want's to have a look at this one.

File: Models/MicroClimate/MicroClimateZone.cs [hunk 1]
```diff
@@ -515,10 +518,15 @@ public void SetCanopyEnergyTerms()
...
+                        totalRsGreen += Canopies[j].Rs[i] * RadnGreenFraction(j);
                     }
                     Canopies[j].Canopy.PotentialEP = totalPotentialEp;
                     Canopies[j].Canopy.WaterDemand = totalPotentialEp;
                     Canopies[j].Canopy.LightProfile = lightProfile;
+                    if (Zone is RectangularZone)
+                        Canopies[j].Canopy.fRadnAllZones = (totalRsGreen * (Zone as Zones.RectangularZone).Width) / (radn * areaWidth);
+                    else 
+                        Canopies[j].Canopy.fRadnAllZones = totalRsGreen  / radn;
                 }
```
Comment (hol353) [inline Models/MicroClimate/MicroClimateZone.cs:530]: I can't see where fRadnAllZones is being used. I also don't understand why it is in ICanopy. Why would the canopy of a plant want to know the radn from all zones?

**Comment by hol353** [comment]
This branch needs to be brought up-to-date. Can you please pull from master?

File: Models/MicroClimate/MicroClimateZone.cs (hunk 1 above)
Comment (HamishBrownPFR) [reply Models/MicroClimate/MicroClimateZone.cs:530]: It is used as a reporting variable.  A tall canopy will intercept radiation from its neighboring zone so for checking things are working properly we calculate fRadnAllZones for both zones and make sure their sum does not exceed 1.  This is currently done in manager scripts

**Comment by HamishBrownPFR** [comment]
I had to move on with work on this branch while waiting for review and have stacked other changes in front of what is here.  I might close this pull request, separate some of the bits on the branch onto smaller pull requests and proceed that way.

**Comment by par456** [comment]
Jenkins got stuck on the validation for this PR a few weeks ago, it gets to the Wheat validation and then just freezes up.

 I'm just going to remove some of the old validations from the jenkins queue so only the latest is there and see if that helps.

**Comment by HamishBrownPFR** [comment]
I have replaced this with a new pull request #8544

**Comment by zur003** [comment]
I've been dying of curiosity for some time now. What the heck does "deropapy" mean anyway? Or if it doesn't mean anything in particular, what is the origin of the name?

**Comment by HamishBrownPFR** [comment]
Dynamic Environmental Response Of Phenology And Potential Yield

**Pull request closed without merging**